generator/README.md

```diff
@@ -0,0 +1,824 @@
+    return [self.store sdl_objectForName:SDLRPCParameterNameMaxNumberRFCOMMPorts ofClass:NSNumber.class error:nil];
+}
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+#### Functions
+
+Functions in iOS are implemented as 3 different classes grouped by their respective type. All the 3 extend the common parent class `SDLRPCMessage`.
+
+// This is the parent class for all the 3 function kinds
+SDLRPCMessage <NSCopying> : SDLRPCStruct <NSCopying> : NSObject
+
+SDLRPCNotification : SDLRPCMessage  /// An RPC sent from the head unit to the app about some data change, such as a button was pressed
+SDLRPCRequest : SDLRPCMessage   /// Superclass of RPC requests
+SDLRPCResponse : SDLRPCMessage  /// Superclass of RPC responses
+```
+
+First of all there is the `SDLFunctionID` class generated though it is not declared in the XML. This class maps all function IDs that are integers to function names as strings.
+
+1. Uses of the `"name"` attribute should be normalized by the removal of the ID suffix, e.g. `RegisterAppInterfaceID -> RegisterAppInterface`. 
+2. The constant name should be camel case formatted.
+0. The constant has 2 fields the first is the `int` value of the `"value"` attribute and the second is the `String` value of normalized `"name"` attribute.
+
+Internally it uses another file that lists all the function names `SDLRPCFunctionNames`.
+
+```objc
+//  SDLFunctionID.h
+
+#import <Foundation/Foundation.h>
+#import "NSNumber+NumberType.h"
+#import "SDLRPCFunctionNames.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// A function ID for an SDL RPC
+@interface SDLFunctionID : NSObject
+
+/// The shared object for pulling function id information
++ (instancetype)sharedInstance;
+
+/// Gets the function name for a given SDL RPC function ID
+///
+/// @param functionID A function ID
+/// @returns An SDLRPCFunctionName
+- (nullable SDLRPCFunctionName)functionNameForId:(UInt32)functionID;
+
+/// Gets the function ID for a given SDL RPC function name
+///
+/// @param functionName The RPC function name
+- (nullable NSNumber<SDLInt> *)functionIdForName:(SDLRPCFunctionName)functionName;
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+Each <function> from MOBILE_API.XML is declares its function name in `SDLRPCFunctionNames.h` and `SDLRPCFunctionNames.m` files.
+
+```objc
+SDLRPCFunctionNames.h
+#import "SDLEnum.h"
+/**
+* All RPC request / response / notification names
+*/
+typedef SDLEnum SDLRPCFunctionName SDL_SWIFT_ENUM;
+
+/// Function name for an AddCommand RPC
+extern SDLRPCFunctionName const SDLRPCFunctionNameAddCommand;
+
+/// Function name for an AddSubMenu RPC
+extern SDLRPCFunctionName const SDLRPCFunctionNameAddSubMenu;
+
+. . . and so on
+```
+
+And the implementation file SDLRPCFunctionNames.m :
+
+```objc
+//
+//  SDLRPCFunctionNames.m
+//  SmartDeviceLink
+//
+
+#import "SDLRPCFunctionNames.h"
+
+SDLRPCFunctionName const SDLRPCFunctionNameAddCommand = @"AddCommand";
+SDLRPCFunctionName const SDLRPCFunctionNameAddSubMenu = @"AddSubMenu";
+
+. . . and so on
+```
+
```

Comment:
This is missing information on `SDLRPCParameterNames`

Reply:
please have look into https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/4b7197ee027bb8b437a427cfdf3bbca01c6cca02